packages/flutter/lib/src/material/ink_decoration.dart

```diff
@@ -0,0 +1,356 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material.dart';
+
+/// A convenience widget for drawing images and other decorations on [Material]
+/// widgets, so that [InkWell] and [InkResponse] splashes will render over them.
+///
+/// Ink splashes and highlights, as rendered by [InkWell] and [InkResponse],
+/// draw on the actual underlying [Material], under whatever widgets are drawn
+/// over the material (such as [Text] and [Icon]s). If an opaque image is drawn
+/// over the [Material] (as with a [Container] or [DecoratedBox]), these ink
```

Comment:
Container and DecoratedBox aren't necessarily opaque, so the unqualified parenthetical statement is a little confusing.

Reply:
replaced with "maybe using a ..."